lib/param_parsing/parameters.go

```diff
@@ -35,6 +35,24 @@ type Params struct {
 	VersionFlag      bool
 }
 
+type paramMapping struct {
+	description string
+	env         string
+	param       string
+	toml        string
+}
+
+// This is used to automatically instate Environment variables and TOML keys
+var paramMappings = []paramMapping{
+	{param: "Arch", env: "TF_ARCH", toml: "arch", description: "CPU architecture"},
+	{param: "CustomBinaryPath", env: "TF_BINARY_PATH", toml: "bin", description: "Custom binary path"},
+	{param: "DefaultVersion", env: "TF_DEFAULT_VERSION", toml: "default-version", description: "Default version"},
+	{param: "InstallPath", env: "TF_INSTALL_PATH", toml: "install", description: "Custom install path"},
+	{param: "LogLevel", env: "TF_LOG_LEVEL", toml: "log-level", description: "Log level"},
+	{param: "Product", env: "TF_PRODUCT", toml: "product", description: "Product"},
```

Comment:
I feel like there's a bunch of overlap with https://github.com/warrensbox/terraform-switcher/pull/566/files#diff-6102d401ce5628c1df4636f2ca1d1d81e8d52346a74662f47d018ed8611b436bR73
Could the whole definitions not be used, so we don't duplicate name, and description and keep the definitions in a single place?

Reply:
Yep, we probably can. Though I don't see this mapping to match the `getopts` thing as this new mapping is solely for parity between env vars and params in TOML.
We can probably expand this new mapping onto the `getops` 🤔 But defo not in scope of this PR which is about another feature. I also do not want to overload this PR with changes that worth of its own issue and PR. Does it sound meaningful?

Reply:
It was mainly the duplicate descriptions that concerned me tbh

Reply:
You talked me round =) Let me push update and see if I got your idea right.